src/Azure.Functions.ArtifactAssembler/ArtifactAssembler.cs

```diff
@@ -161,12 +198,150 @@ private async Task CreateVisualStudioCoreToolsAsync()
+                    inProc8ArtifactDirPath = inProc8Response[0];
+                    inProcVersion = inProc8Response[1];
+                }
+                else
+                {
+                    var artifactNameWithVersion = $"{artifactName}.{inProcVersion}";
+                    inProc8ArtifactDirPath = Path.Combine(_inProc8ExtractedRootDir, artifactNameWithVersion);
+                }
+
+                // Rename inproc8 directory to have the same version as the out-of-proc artifact before copying
+                string newInProc8ArtifactDirPath = RenameInProcDirectory(inProc8ArtifactDirPath, outOfProcVersion);
+
+                // Copy in-proc8 files
+                var inProc8CopyTask = Task.Run(() => FileUtilities.CopyDirectory(newInProc8ArtifactDirPath, Path.Combine(consolidatedArtifactDirPath, InProc8DirectoryName)));
+
+                // Rename inproc6 directory to have the same version as the out-of-proc artifact before copying
+                var inProcArtifactName = Path.GetFileName(inProc8ArtifactDirPath);
+                var inProc6ArtifactDirPath = Path.Combine(_inProc6ExtractedRootDir, inProcArtifactName);
+                EnsureArtifactDirectoryExist(inProc6ArtifactDirPath);
+                string newInProc6ArtifactDirPath = RenameInProcDirectory(inProc6ArtifactDirPath, outOfProcVersion);
+
+                // Copy in-proc6 files
+                var inProc6CopyTask = Task.Run(() => FileUtilities.CopyDirectory(newInProc6ArtifactDirPath, Path.Combine(consolidatedArtifactDirPath, InProc6DirectoryName)));
+
+                await Task.WhenAll(inProc8CopyTask, inProc6CopyTask, outOfProcCopyTask);
+
+                // consolidatedArtifactDirPath now contains custom core-tools host, in-proc6 and in-proc8 sub directories. Create a zip file.
+                var zipPath = Path.Combine(cliCoreToolsTargetArtifactDir, $"{outOfProcArtifactName}.zip");
+                try
+                {
+                    await Task.Run(() => FileUtilities.CreateZipFile(consolidatedArtifactDirPath, zipPath));
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine(ex.ToString());
+                }
+                Console.WriteLine($"Successfully created target runtime zip at: {zipPath}");
+
+                Directory.Delete(consolidatedArtifactDirPath, true);
+            });
+
             await Task.WhenAll(packTasks);
 
             // Delete the extracted directories
             Directory.Delete(_inProc6ExtractedRootDir, true);
             Directory.Delete(_inProc8ExtractedRootDir, true);
             Directory.Delete(_coreToolsHostExtractedRootDir, true);
+            Directory.Delete(_outOfProcExtractedRootDir, true);
+
+            Console.WriteLine($"Finished assembling CLI Core Tools artifacts");
+        }
+
+        private string[] GetArtifactDirectoryAndVersionNumber(string extractedRootDirectory, string artifactName)
```

Comment:
Can you rebase? We have a method to get this information now.

https://github.com/Azure/azure-functions-core-tools/blob/df4c712092e425fff63bb30b7e4231984bef5a02/src/Azure.Functions.ArtifactAssembler/ArtifactAssembler.cs#L127-L136

Reply:
If you still need to use this method, `string[]` may not be the appropriate return type here. Either use named tuple or a POCO to improve readability.